core/src/main/scala/kafka/security/auth/SimpleAclAuthorizer.scala

```diff
@@ -87,6 +94,17 @@ class SimpleAclAuthorizer extends Authorizer with Logging {
       case str: String if str.nonEmpty => str.split(",").map(s => KafkaPrincipal.fromString(s.trim)).toSet
     }.getOrElse(Set.empty[KafkaPrincipal])
 
+    def getInstanceForConfig(configName: String): Option[Nothing] = {
+      configs.get(configName).collect {
+        case str: String if str.nonEmpty => CoreUtils.createObject(str)
+      }
+    }
+    principalToLocalPlugin = getInstanceForConfig(SimpleAclAuthorizer.PrincipalToLocalProp)
```

Comment:
@ijuma Thanks for pointing that out. KerberosShortNamer seems specific to SASL.

I think if we can make KerberosShortNamer implement some sort of interface (may be the one specified in this PR) that users can inject that will be more extensible and then we can also reuse that same interface for Authorizer. I can do that work if we agree on that.